### PR TITLE
Make boosted_trees Garden-official

### PR DESCRIPTION
**Pull request by yhliang2018**
Hi All,

I update the boosted_trees code to make it more garden-style:
1. Add official flags in data_download.py, and fix a minor bug
2. Add benchmark logger in train_higgs.py
3. Update single quote with double quotes

### CODE REVIEW

**Comment by yhliang2018** [comment]
@yk5 Could you help to review the code? Thanks!

File: official/boosted_trees/data_download.py [hunk 1]
```diff
@@ -73,6 +69,16 @@ def main(unused_argv):
   _download_higgs_data_and_save_npz(FLAGS.data_dir)
 
 
-if __name__ == '__main__':
-  FLAGS, unparsed = parse_args()
-  tf.app.run(argv=[sys.argv[0]] + unparsed)
+def define_data_download_flags():
+  """Add flags specifying data download arguments."""
```
[the remark this answers is not in the document]
Comment (yhliang2018) [reply official/boosted_trees/data_download.py]: Good point! @robieta Maybe we should add one in utils/flags?

File: official/boosted_trees/train_higgs.py [hunk 2]
```diff
@@ -29,64 +29,43 @@
 from __future__ import division
 from __future__ import print_function
 
-import argparse
 import os
-import sys
 
+# pylint: disable=g-bad-import-order
+import numpy as np
```
[the remark this answers is not in the document]
Comment (yhliang2018) [reply official/boosted_trees/train_higgs.py]: I got lint errors in Kokoro checking if numpy goes after absl. :(

File: official/boosted_trees/data_download.py [hunk 3]
```diff
@@ -12,59 +12,55 @@
...
 
     # Reading and parsing 11 million csv lines takes 2~3 minutes.
-    print('Data processing.. taking multiple minutes..')
-    data = pd.read_csv(
-        temp_filename,
-        dtype=np.float32,
-        names=['c%02d' % i for i in range(29)]  # label + 28 features.
-    ).as_matrix()
+    tf.logging.info("Data processing... taking multiple minutes...")
+    with gzip.open(temp_filename, "rb") as csv_file:
```
[the remark this answers is not in the document]
Comment (yhliang2018) [reply official/boosted_trees/data_download.py]: Thanks for pointing it out! It's strange then, as when I tested the original code, I got the following error:
```
pandas.errors.ParserError: Error tokenizing data. C error: Buffer overflow caught - possible malformed input file.
```
That's why we explicitly gzip it here. Any idea on the issue?

[the remark this answers is not in the document]
Comment (yk5) [reply official/boosted_trees/data_download.py]: Maybe related to pandas version, but as gzip works, I think this change is fine.

[the remark this answers is not in the document]
Comment (yhliang2018) [reply official/boosted_trees/data_download.py]: Which version are you using? I use 0.22.0.

[the remark this answers is not in the document]
Comment (yk5) [reply official/boosted_trees/data_download.py]: Hmm. the same 0.22.0. Do you get errors when running locally in virtualenv? or in travis or whatever?
FYI, I'm using Linux with virtualenv (python 2.7.13 numpy 1.14.3).
I ran it just now and confirmed pd.read_csv() reads and processes the csv.gz file properly..

[the remark this answers is not in the document]
Comment (yhliang2018) [reply official/boosted_trees/data_download.py]: Aha, I see the problem. I ran it with python3. When I test it with python2, it works well as yours. So I will just keep gzip explicitly for py2 and py3 compatibility. Thanks a lot! :)

[the remark this answers is not in the document]
Comment (yk5) [reply official/boosted_trees/data_download.py]: I see. Thanks for the fix!

**Pull request merged**